tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -168,13 +152,13 @@ def test_non_ocs_taint_and_tolerations(self, nodes):
             self.sanity_helpers.health_check()
 
         logger.info("Check number of pods before and after adding non ocs taint")
-        number_of_pods_after = len(
-            get_all_pods(namespace=config.ENV_DATA["cluster_namespace"])
-        )
-        assert (
-            number_of_pods_before == number_of_pods_after
-        ), "Number of pods didn't match"
-
+        retry(
```

Comment:
it would be much better to use retry as function decorator - current code is hard to read and maintain